akka-contrib/src/main/scala/akka/contrib/pattern/ReceivePipeline.scala

```diff
@@ -38,10 +38,13 @@ trait ReceivePipeline extends Actor {
     val around = aroundCache match {
       case Some((`receive`, cached)) ⇒ cached
       case _ ⇒
-        val zipped = pipeline.foldRight[Receive](receive)((outer, inner) ⇒ outer(inner).orElse[Any, Unit](inner))
+        val zipped = pipeline.foldRight[Receive] {
+          case msg => super.aroundReceive(receive, msg)
```

Comment:
When @ktoso suggested that `super.aroundReceive` should be called somewhere, I agreed: what right has `ReceivePipeline` to enforce the specific use of `unhandled` as the fallback when the base actor might deliberately have overridden `aroundReceive` with some other behaviour. 

Placing the `super.aroundReceive` call in the middle of the fold makes the fallback less invasive. But now as you point out, @ktonga, it might not get called at all. Clearly, calling `super.aroundReceive` at the end of the method _as well_ will not do as then it will then often be called twice.

My gut feeling is that it is almost as if `ReceivePipeline` is ceding too much power over the flow control to the interceptors (allowing them to call each other directly). An alternative would be to have each interceptor as a `PartialFunction[Any, InterceptorResult]` with something like:
```
sealed trait InterceptorResult
case class Inner(transformedMsg: Any) extends InterceptorResult
case object HandledCompletely extends InterceptorResult
```
The ReceivePipeline would construct a `Receive` from these interceptors (so that _it_ would be responsible for the message flow between interceptors) and just pass this to `super.aroundReceive` along with the original message at the end (as per the original implementation). 

Of course this would be an API changer.